### PR TITLE
feat: Created kubernetes service account module

### PR DESCRIPTION
**Pull request by diegolagospagopa**
### Motivation and context

Allow to create a service account with default secret

### Type of changes

- [ ] Add new module
- [ ] Update existing module
- [ ] Remove existing module

### Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### Run checks

Useful commands to run checks on local machine

```sh
bash .utils/terraform_run_all.sh init local
pre-commit run -a
```

### CODE REVIEW

**Comment by Krusty93** [comment]
What's the scope of this module? README.md mentions storage accounts but main.tf has a Kube Secrets referencing an Azure DevOps service account?

File: kubernetes_service_account/versions.tf [hunk 1]
```diff
@@ -0,0 +1,10 @@
+terraform {
+  required_version = ">= 1.3.0"
+
+  required_providers {
+    kubernetes = {
+      source  = "hashicorp/kubernetes"
+      version = ">= 2.10.0"
```
Comment (diegolagospagopa) [inline kubernetes_service_account/versions.tf]:
```suggestion
      version = ">= 2.25.0"
```

**Comment by diegolagospagopa** [comment]
> What's the scope of this module? README.md mentions storage accounts but main.tf has a Kube Secrets referencing an Azure DevOps service account?

The readme was old, now is updated.
This module allow to create a service account with token and ca.crt for azure devops purpose

**Comment by Krusty93** [comment]
> > What's the scope of this module? README.md mentions storage accounts but main.tf has a Kube Secrets referencing an Azure DevOps service account?
> 
> The readme was old, now is updated. This module allow to create a service account with token and ca.crt for azure devops purpose

Wouldn't be better to have something reminding DevOps in the module name? As it is it looks like a wrapper around `kubernetes_service_account`

**Pull request merged**